tree/tree/src/TBasket.cxx

```diff
@@ -721,17 +788,26 @@ void TBasket::Reset()
    if (newNevBufSize==0) {
       delete [] fEntryOffset;
       fEntryOffset = 0;
+      delete [] fCompressedEntryOffset;
+      fCompressedEntryOffset = 0;
    } else if (newNevBufSize != fNevBufSize) {
       delete [] fEntryOffset;
       fEntryOffset = new Int_t[newNevBufSize];
-   } else if (!fEntryOffset) {
-      fEntryOffset = new Int_t[newNevBufSize];
-   }
+      delete [] fCompressedEntryOffset;
+      fCompressedEntryOffset = new Int_t[newNevBufSize];
+   } else if (!fEntryOffset || !fCompressedEntryOffset) {
+      if (!fEntryOffset)
+         fEntryOffset = new Int_t[newNevBufSize];
+      if (!fCompressedEntryOffset)
```

Comment:
This gives the impression that fCompressedEntryOffset is always created.  It should only cost in memory only when used.